test/unit/init.test.js

```diff
@@ -0,0 +1,373 @@
+            Passwordless.getLinkCodeFromURL();
+        } catch (err) {
+            assert(
+                err.message.startsWith(
+                    "No instance of Passwordless found. Ensure that the 'Passwordless.init' method is called within the 'SuperTokens.init' recipeList."
+                )
+            );
+        }
+    });
+
+    it("Throws correct error when calling Passwordless methods if recipe is not initialized but SuperTokens is initialized", async function () {
+        SuperTokens.init({
+            appInfo: {
+                appName: "SuperTokens",
+                apiDomain: "api.supertokens.io",
+            },
+            windowHandler: function (original) {
+                return {
+                    ...original,
+                    location: {
+                        ...original.location,
+                        getHostName: () => {
+                            return "http://localhost:3000";
+                        },
+                    },
+                };
+            },
+            recipeList: [EmailPassword.init()],
+        });
+        try {
+            Passwordless.getLinkCodeFromURL();
+        } catch (err) {
+            assert(
+                err.message.startsWith(
+                    "No instance of Passwordless found. Ensure that the 'Passwordless.init' method is called within the 'SuperTokens.init' recipeList."
+                )
+            );
+        }
+    });
+
+    it("Throws correct error when calling Session methods if SuperTokens is not initialized", async function () {
+        try {
+            await Session.getUserId();
+        } catch (err) {
+            assert(
+                err.message.startsWith(
+                    "No instance of Session found. Ensure that the 'Session.init' method is called within the 'SuperTokens.init' recipeList."
+                )
+            );
+        }
+    });
+
+    it("Throws correct error when calling Session methods if recipe is not initialized but SuperTokens is initialized", async function () {
+        SuperTokens.init({
+            appInfo: {
+                appName: "SuperTokens",
+                apiDomain: "api.supertokens.io",
+            },
+            windowHandler: function (original) {
+                return {
+                    ...original,
+                    location: {
+                        ...original.location,
+                        getHostName: () => {
+                            return "http://localhost:3000";
+                        },
+                    },
+                };
+            },
+            recipeList: [EmailPassword.init()],
+        });
+        try {
+            await Session.getUserId();
+        } catch (err) {
+            assert(
+                err.message.startsWith(
+                    "No instance of Session found. Ensure that the 'Session.init' method is called within the 'SuperTokens.init' recipeList."
+                )
+            );
+        }
```

Comment:
Extracting these checks into a single util function that takes a callback and the recipe name would make this a bit easier to read.

Reply:
Ignoring theses as discussed